catalog/core/catalog-core-api-impl/src/main/java/ddf/catalog/content/data/impl/ContentItemImpl.java

```diff
@@ -0,0 +1,162 @@
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+
+import javax.activation.MimeType;
+import javax.activation.MimeTypeParseException;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.google.common.io.ByteSource;
+
+import ddf.catalog.content.data.ContentItem;
+import ddf.catalog.data.Metacard;
+
+public class ContentItemImpl implements ContentItem {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ContentItemImpl.class);
+
+    protected String id;
+
+    protected String uri;
+
+    protected String filename;
+
+    protected String mimeTypeRawData;
+
+    protected MimeType mimeType;
+
+    protected ByteSource byteSource;
+
+    protected File file;
+
+    protected long size;
+
+    protected Metacard metacard;
+
+    /**
+     * An incoming content item since the ID will initially be
+     * <code>null</code> because the {@link ddf.catalog.CatalogFramework} will assign its GUID.
+     *
+     * @param byteSource          the {@link ContentItem}'s input stream containing its actual data
+     * @param mimeTypeRawData the {@link ContentItem}'s mime type
+     */
+    public ContentItemImpl(ByteSource byteSource, String mimeTypeRawData, String filename,
+            Metacard metacard) {
+        this(null, byteSource, mimeTypeRawData, filename, 0, metacard);
+    }
+
+    /**
+     * An incoming content item where the item's GUID should be known.
+     *
+     * @param id              the {@link ContentItem}'s GUID
+     * @param byteSource          the {@link ContentItem}'s input stream containing its actual data
+     * @param mimeTypeRawData the {@link ContentItem}'s mime type
+     */
+    public ContentItemImpl(String id, ByteSource byteSource, String mimeTypeRawData,
+            Metacard metacard) {
+        this(id, byteSource, mimeTypeRawData, null, 0, metacard);
+    }
+
+    public ContentItemImpl(String id, ByteSource byteSource, String mimeTypeRawData, String filename, long size,
+            Metacard metacard) {
+        this.byteSource = byteSource;
+        this.mimeTypeRawData = mimeTypeRawData;
+        this.id = id;
+        this.mimeType = null;
+        this.size = size;
+        if (filename != null) {
+            this.filename = filename;
+        } else {
+            this.filename = DEFAULT_FILE_NAME;
+        }
+        this.metacard = metacard;
+
+        if (mimeTypeRawData != null) {
+            try {
+                this.mimeType = new MimeType(mimeTypeRawData);
+            } catch (MimeTypeParseException e) {
+                LOGGER.debug("Unable to create MimeType from raw data " + mimeTypeRawData);
+            }
+        } else {
+            this.mimeTypeRawData = DEFAULT_MIME_TYPE;
+            try {
+                this.mimeType = new MimeType(this.mimeTypeRawData);
+            } catch (MimeTypeParseException e) {
+                LOGGER.debug("Unable to create MimeType from raw data " + this.mimeTypeRawData);
+            }
```

Comment:
Redundant `try` blocks - I would move the entire `if` statement into a single `try-catch`.